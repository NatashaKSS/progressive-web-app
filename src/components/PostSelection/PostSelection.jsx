import React from 'react';
import { Link } from 'react-router-dom';

import styles from './PostSelection.css';

import PostsAPI from '../../api/PostsAPI.js';

const PostSelection = () => {
  return (
    <div>
      <ul>
        {PostsAPI.getAll().map(post => {
          return (
            <li className={styles.postItem} key={post.id}>
              <Link to={`/posts/${post.id}`}>{post.title}</Link>
            </li>
          );
        })}
      </ul>
    </div>
  );
}

export default PostSelection;
