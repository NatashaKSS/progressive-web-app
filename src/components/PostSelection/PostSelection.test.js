import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PostSelection from './PostSelection';

test('<PostSelection /> should render as expected', () => {
  const component = shallow(<PostSelection />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
