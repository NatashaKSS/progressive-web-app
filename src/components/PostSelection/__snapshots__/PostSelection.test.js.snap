// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostSelection /> should render as expected 1`] = `
<div>
  <ul>
    <li
      className="postItem"
    >
      <Link
        replace={false}
        to="/posts/010"
      >
        Apples
      </Link>
    </li>
    <li
      className="postItem"
    >
      <Link
        replace={false}
        to="/posts/011"
      >
        Salmon
      </Link>
    </li>
    <li
      className="postItem"
    >
      <Link
        replace={false}
        to="/posts/012"
      >
        Pistachios
      </Link>
    </li>
  </ul>
</div>
`;
